tests/test_split_command.rs

```diff
@@ -80,4 +80,99 @@ fn test_split_by_paths() {
     A file1
     A file3
     "###);
+
+    // Insert an empty commit after @- with "split ."
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(
+        edit_script,
+        ["dump editor0", "next invocation\n", "dump editor1"].join("\0"),
+    )
+    .unwrap();
+    let stdout = test_env.jj_cmd_success(&repo_path, &["split", "-r", "@-", "."]);
+    insta::assert_snapshot!(stdout, @r###"
+    Rebased 1 descendant commits
+    First part: 31425b568fcf (no description set)
+    Second part: af0963926ac3 (no description set)
+    Working copy now at: 28d4ec20efa9 (no description set)
+    "###);
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor0")).unwrap(), @r###"
+    JJ: Enter commit description for the first part (parent).
+
+    JJ: This commit contains the following changes:
+    JJ:     A file2
+
+    JJ: Lines starting with "JJ: " (like this one) will be removed.
+    "###);
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor1")).unwrap(), @r###"
+    JJ: Enter commit description for the second part (child).
+
+    JJ: This commit contains the following changes:
+
+    JJ: Lines starting with "JJ: " (like this one) will be removed.
+    "###);
+
+    let stdout =
+        test_env.jj_cmd_success(&repo_path, &["log", "-T", r#"commit_id.short() " " empty"#]);
+    insta::assert_snapshot!(stdout, @r###"
+    @ 28d4ec20efa9 false
+    o af0963926ac3 true
+    o 31425b568fcf false
+    o 000000000000 true
+    "###);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diff", "-s", "-r", "@--"]);
+    insta::assert_snapshot!(stdout, @r###"
+    A file2
+    "###);
+
+    // Remove newly created empty commit
+    test_env.jj_cmd_success(&repo_path, &["abandon", "@-"]);
+
+    // Insert an empty commit before @- with "split nonexistent"
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(
+        edit_script,
+        ["dump editor0", "next invocation\n", "dump editor1"].join("\0"),
+    )
+    .unwrap();
+    let stdout = test_env.jj_cmd_success(&repo_path, &["split", "-r", "@-", "nonexistent"]);
```

Comment:
We may eventually produce an error or warning about the nonexistent path. I'm not sure what to do about this test case then. That makes it extra useful to have as a reminder :)